src/main/java/org/prebid/server/bidder/nextmillenium/NextMilleniumBidder.java

```diff
@@ -0,0 +1,135 @@
+package org.prebid.server.bidder.nextmillenium;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.request.nextmillenium.ExtImpNextMillenium;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class NextMilleniumBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpNextMillenium>> NEXTMILLENIUM_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpNextMillenium>>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public NextMilleniumBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public final Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<Imp> imps = bidRequest.getImp();
+        final List<BidderError> errors = new ArrayList<>();
+        final List<ExtImpNextMillenium> extImps = CollectionUtils.isNotEmpty(imps)
+                ? getImpressionInfo(imps, errors)
+                : errorIfEmpty(errors);
+
```

Comment:
We can update this to this, and also I would suggest to extract ext resolving logic to another method
```java
        final List<BidderError> errors = new ArrayList<>();
        final List<ExtImpNextMillenium> impExts = new ArrayList<>();

        for (Imp imp : bidRequest.getImp()) {
            try {
                impExts.add(mapper.mapper()
                        .convertValue(imp.getExt(), NEXTMILLENIUM_EXT_TYPE_REFERENCE)
                        .getBidder());
            } catch (IllegalArgumentException e) {
                errors.add(BidderError.of(e.getMessage(), BidderError.Type.bad_input));
            }
        }
```